src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1472,15 +1472,19 @@ reg_t MPS::sample_measure_using_probabilities_internal(const rvector_t &rnds,
     return samples;
 }
 
-reg_t MPS::apply_measure(const reg_t &qubits, RngEngine &rng) {
-  // since input is always sorted in qasm_controller, therefore, we must return the qubits 
-  // to their original location (sorted)
-  move_all_qubits_to_sorted_ordering();
-  return apply_measure_internal(qubits, rng);
+reg_t MPS::apply_measure(const reg_t &qubits, RngEngine &rng, bool is_sorted) {
+  // Move qubits in the MPS structure to their original location
+  // unless we know they were already sorted
+  if (!is_sorted)
+    move_all_qubits_to_sorted_ordering();
+  
+  return apply_measure_internal(qubits, rng, is_sorted);
 }
 
+// The caller to apply_measure_internal is responsible to call 
+// move_all_qubits_to_sorted_ordering before calling this function
 reg_t MPS::apply_measure_internal(const reg_t &qubits, 
-				  RngEngine &rng) {
+				  RngEngine &rng, bool is_sorted) {
```

Comment:
It will be clearer and more consistent to remove the `is_sorted` parameter, and make it the caller's responsibility to sort `qubits` as well.

Reply:
Okay, I did as you suggested.